streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextImpl.java

```diff
@@ -214,18 +214,13 @@ public long streamTime() {
         return streamTimeSupplier.get();
     }
 
-    private abstract static class StateStoreReadOnlyDecorator<T extends StateStore> extends AbstractStateStore {
+    private abstract static class StateStoreReadOnlyDecorator<T extends StateStore> extends WrappedStateStore<T> {
         static final String ERROR_MESSAGE = "Global store is read only";
 
         private StateStoreReadOnlyDecorator(final T inner) {
             super(inner);
         }
 
-        @SuppressWarnings("unchecked")
-        T getInner() {
-            return (T) wrappedStore();
-        }
-
```

Comment:
These no-longer-necessary methods demonstrate the benefit of adding the generic parameter to WrappedStateStore. Previously, `wrappedStore` could only return a `StateStore`, but now it directly returns a `T extends StateStore`, with no need for casting anywhere.